File: app/src/test/java/com/tminus1010/buva/core_testing/Log.java
package android.util;
// Intentionally changed this^ so that Log doesn't cause unit-test crashes.

public class Log {
    public static int d(String tag, String msg) {
        System.out.println(msg);
        return 0;
    }

    public static int i(String tag, String msg) {
        System.out.println(msg);
        return 0;
    }

    public static int w(String tag, String msg) {
        System.out.println(msg);
        return 0;
    }

    public static int e(String tag, String msg) {
        System.out.println(msg);
        return 0;
    }
}

File: app/src/test/java/com/tminus1010/buva/core_testing/Log.kt
package android.util
// Intentionally changed this^ package to override android.util.Log during unit-tests.

object Log {
    @JvmStatic
    fun d(tag: String?, msg: String?): Int {
        println(msg)
        return 0
    }

    @JvmStatic
    fun i(tag: String?, msg: String?): Int {
        println(msg)
        return 0
    }

    @JvmStatic
    fun w(tag: String?, msg: String?): Int {
        println(msg)
        return 0
    }

    @JvmStatic
    fun e(tag: String?, msg: String?): Int {
        println(msg)
        return 0
    }
}